lib/src/rules/join_return_with_assignment.dart

```diff
@@ -35,15 +35,15 @@ class A {
 
 ''';
 
-Element _getElementFromAssignmentStatement(Statement node) {
+Expression _getExpressionFromAssignmentStatement(Statement node) {
   final visitor = new _AssignmentStatementVisitor();
   node.accept(visitor);
-  return visitor.element;
+  return visitor.expression;
 }
 
-Element _getElementFromReturnStatement(Statement node) {
+Expression _getExpressionFromReturnStatement(Statement node) {
```

Comment:
Perhaps consider ternary and short-hand:

    => node is ReturnStatement ? node.expression : null;

?

Reply:
Done.